src/frontend/hooks/useNavigationWithTypes.ts

```diff
@@ -0,0 +1,19 @@
+import { BottomTabNavigationProp } from "@react-navigation/bottom-tabs";
+import {
+  CompositeNavigationProp,
+  useNavigation,
+} from "@react-navigation/native";
+import { NativeStackNavigationProp } from "@react-navigation/native-stack";
+import { AppStackList, HomeTabsList } from "../Navigation/AppStack";
+
+export const useNavigationFromRoot = () =>
+  useNavigation<NativeStackNavigationProp<AppStackList>>();
+
+export function useNavigationFromHomeTabs() {
+  return useNavigation<
+    CompositeNavigationProp<
+      BottomTabNavigationProp<HomeTabsList>,
+      NativeStackNavigationProp<AppStackList>
+    >
+  >();
+}
```

Comment:
In simplest terms, the use navigation hooks takes the screen names as a type in order to provide type-safety when using the hooks. So when you use the `useNavigationFromRoot`, you can only navigate to a screen in `AppStackList`.

The reason we have 2 is for the nested navigators. If your in the home tabs: you can either navigate to `camera`, `map` OR `root, {screen:screen-name}` screen. Since the root is nested, this second hook just makes it so the developer has to explicitly code that they are navigating into a nested navigator (`root`) and which screen they are navigating into